Marlin/src/sd/cardreader.cpp

```diff
@@ -472,46 +472,48 @@ void CardReader::mount() {
 #endif
 
 void CardReader::manage_media() {
-  static uint8_t prev_stat = 2;     // First call, no prior state
-  uint8_t stat = uint8_t(IS_SD_INSERTED());
-  if (stat == prev_stat) return;
+  static struct { bool present:1, inited:1; } media_stat = { false, false };
 
-  DEBUG_SECTION(mm, "CardReader::manage_media", true);
-  DEBUG_ECHOLNPGM("SD Status ", prev_stat, " -> ", stat);
+  bool is_present = IS_SD_INSERTED();
+  if (is_present == media_stat.present) return; // No change in media presence?
 
-  flag.workDirIsRoot = true;        // Return to root on mount/release
+  DEBUG_SECTION(cmm, "CardReader::manage_media()", true);
+  DEBUG_ECHOLNPGM("Media present: ", media_stat.present, " -> ", is_present);
+
+  flag.workDirIsRoot = true;                  // Return to root on mount/release/init
 
   if (!ui.detected()) {
     DEBUG_ECHOLNPGM("SD: No UI Detected.");
     return;
   }
 
-  uint8_t old_stat = prev_stat;
-  prev_stat = stat;                 // Change now to prevent re-entry
+  const bool was_present = media_stat.present;
+  media_stat.present = is_present;
 
-  if (stat) {                       // Media Inserted
-    safe_delay(500);                // Some boards need a delay to get settled
+  if (is_present) {                           // Media Inserted
+    safe_delay(500);                          // Some boards need a delay to get settled
 
     // Try to mount the media (only later with SD_IGNORE_AT_STARTUP)
     if (TERN1(SD_IGNORE_AT_STARTUP, media_stat.inited)) mount();
-    if (!isMounted()) stat = 0;     // Not mounted?
+    if (!isMounted()) is_present = false;     // Not mounted?
 
     TERN_(RESET_STEPPERS_ON_MEDIA_INSERT, reset_stepper_drivers()); // Workaround for Cheetah bug
   }
   else {
     #if PIN_EXISTS(SD_DETECT)
-      release();                    // Card is released
+      release();                              // Card is released
     #endif
   }
 
-  ui.media_changed(old_stat, stat); // Update the UI
+  ui.media_changed(was_present, is_present);  // Update the UI or flag an error
```

Comment:
Note that `ui.media_changed` throws an error if the two parameters are equal, assuming that a `mount` or `unmount` failed. The values were not the same in previous code during init because `prev_stat` was defined as 2.